### DISCUSSION TITLE
Set connection configuration option

### DISCUSSION THREAD

**Discussion by JamesDrummond** [eclipse-theia/theia#9497] [Q&A]
How do I set connection configuration options such as https://github.com/theia-ide/theia/blob/master/packages/core/src/browser/messaging/connection.ts#L81 ?

**Comment by epatpol** [comment]
Right now it's not an option I believe, although it was discussed beforehand that maybe it should.

**Comment by JamesDrummond** [comment]
@epatpol Maybe set it via environment variables. This seems to be a the easiest solution. If later a configuration file is needed it would just globally set environment variable if not set already. 

```
const maxReconnectionDelay= GetEnvironmentVar("THEIA_MAXRECONNECTIONDELAY",10000);
.....
            const socketOptions = {
                maxReconnectionDelay: maxReconnectionDelay,
.....
            };
```

**Comment by epatpol** [comment]
Yes, we actually already have an env variable server which can be used to easily fetch the variable value from `connection.ts` [here](https://github.com/theia-ide/theia/blob/aef8737c78bc499ae156e9837895ef7b0d0b4e04/packages/core/src/common/env-variables/env-variables-protocol.ts). PRs are welcome :)

**Comment by JamesDrummond** [comment]
@epatpol How do I use the environment server?

**Comment by epatpol** [comment]
You need to inject it in the `WebSocketConnectionProvider` and use it afterwards. Look at how we inject an instance of `ILogger` in `KeybindingRegistry` for instance.

**Comment by JamesDrummond** [comment]
@epatpol Thanks for pointing me in the right direction. Think it will be like https://github.com/theia-ide/theia/blob/master/packages/languages/src/browser/language-client-factory.ts#L34. See what I can do for a PR.

```
@inject(WebSocketConnectionProvider) protected readonly connectionProvider: WebSocketConnectionProvider
...
connectionProvider: {
                get: (errorHandler: ConnectionErrorHandler, closeHandler: ConnectionCloseHandler) =>
                    new Promise(resolve => {
                        this.connectionProvider.listen({
                            path: envVariablesPath ,
                            onConnection: messageConnection => {
                                const connection = createConnection(messageConnection, errorHandler, closeHandler);
                                resolve(connection);
                            }
                        },
                            { reconnecting: false }
                        );
                    })
            }
```

**Comment by svenefftinge** [comment]
Aren't those numbers too high anyway? It's good to be able to configure it through envs but we should have good defaults, too. @JamesDrummond what is the reason you want to change them?
